internal/services/order/order.go

```diff
@@ -1,13 +1,212 @@
 package order
 
-func BuyOrder(BReq BuyRequest) (bRes BuyResponse, err error) {
-	return bRes, err
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"Tradeasy/internal/services/stock_exchange"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+func BuyOrder(bReq BuyRequest) (bRes stock_exchange.OrderResponse, err error) {
+	var checkLtp model.StocksFeed
+	var balance model.Payments
+
+	if bReq.BookType == "Market" {
+		if err = config.DB.Table("stocks_feed").Where("stock_name=?", bReq.StockName).Last(&checkLtp).Error; err != nil {
+			return bRes, err
+		}
+		bReq.LimitPrice = checkLtp.LTP
+	}
+
+	if err = config.DB.Table("payments").Where("user_id=?", bReq.UserId).First(&balance).Error; err != nil {
+		return bRes, err
+	} else if balance.CurrentBalance < int64(bReq.Quantity*bReq.LimitPrice) {
+		return bRes, errors.New("balance is insufficient for the placed order")
+	} else {
+		balance.CurrentBalance = balance.CurrentBalance - int64(bReq.Quantity*bReq.LimitPrice)
```

Comment:
No need to put it in else block. Write it independent of if else. if code passes the if elses then that means balance is to be deducted.